!!! TASK: Modernized code style in ReflectionService

robertlemke opened the pull request:
Code in the reflection service was adjusted to the current code style best practices. 

The method arguments in the Reflection Service are now strictly typed. Therefore, third-party code which relied on loose types and passes invalid types, need to be adjusted. Tests in the Flow package were adjusted were necessary.

As part of the clean up, the setStatusCache() method in ReflectionService was fixed which used a wrong order of parameters in its is_callable() call.

Preparation for #2913

robertlemke committed:
TASK: Adjust code style in ReflectionService

#2913

albe commented:
Can we just merge this as is (after fixing the builds) to keep the following functional changes to the ReflectionService away from the noise of these (useful! 👏) syntax changes? :)

mhsdesign commented on Neos.Flow/Classes/Reflection/ReflectionService.php:
```diff
@@ -13,27 +14,33 @@
 use Doctrine\Persistence\Proxy as DoctrineProxy;
-use Neos\Flow\Annotations as Flow;
-use Neos\Cache\Frontend\FrontendInterface;
+use InvalidArgumentException;
```
i think you configured your ide like this...

i dont use it myself and it is an anomaly to the rest of neos/flow
just wanted to start the diskussion.

do we want this for new code?

robertlemke replied:
Hmm, I don't know … I like it and do that for a couple of years now in my own code. It's the easiest rule: Import if possible.

kdambekalns replied:
Good point. I dislike this and exclude the global namespace from being imported in PhpStorm.

For me imports like these do not add anything useful.

kdambekalns replied:
To me this usually adds a feeling of "oh, something custom here…" – as in, not built-in into PHP itself. 🤷‍♂️

But the "one rule to rule them all" is a point… I'd follow suit if we decide to do it.

mhsdesign replied:
slack has your back @kdambekalns https://neos-project.slack.com/archives/C04Q0TC15/p1683558099679779 😎

mhsdesign commented:
i agree with @albe this syntax and type adjustment is already a lot to review ... if its non breaking it can go directly to 8.2

robertlemke committed:
TASK: Adjust tests to strict types in ReflectionService

The ReflectionService uses strict type declarations in its method calls now, therefore, a couple of tests related to commands and the Type Converter were adjusted.

robertlemke committed:
BUGFIX: Fix ReflectionService::setStatusCache()

The setStatusCache() method in ReflectionService was fixed which used a wrong order of parameters in its is_callable() call.

kdambekalns commented:
> if its non breaking it can go directly to 8.2

It could, but keeping the non-breaking tweaks seperated from the (potentially) breaking changes is adding a lot of overhead during development. And eventually something will break. So let's stick to the 9.0 branch for this!

kdambekalns commented on Neos.Flow/Classes/Reflection/ReflectionService.php:
```diff
@@ -627,10 +510,10 @@ public function isClassAbstract($className)
      *
      * @param string $className Name of the class to analyze
-     * @return boolean true if the class is final, otherwise false
+     * @return bool true if the class is final, otherwise false
```
This could be left out, as the comment does not add information that's not already in the name…

robertlemke replied:
> This could be left out, as the comment does not add information that's not already in the name…

well, yes … but I guess the ReflectionService will change a lot anyway during the next days …

kdambekalns reviewed (commented):
Maybe take the "global namespace import" issue to the team slack for a quick vote?

robertlemke commented:
Sorry for the force-pushes … The state should now be fine, I removed the global import of `\InvalidArgumentException` as the general consensus in the team is to rather not import such classes.

kdambekalns commented:
Regarding the Psalm failures, the one about Fluid might need serious checking, that could be true.

The others _might_ be false positives, but maybe when the refactoring has reached "the next level" we could tackle what is left…

robertlemke commented:
Thanks for your reviews!
